browser/ui/sidebar/sidebar_controller.cc

```diff
@@ -237,17 +257,34 @@ void SidebarController::OnTabStripModelChanged(
           sidebar_model_->active_index() != wanted_index.value()) {
         // Don't pass through null values, as we don't want to close existing
         // panels if we're not opening a tab-specific panel.
-        ActivateItemAt(wanted_index.value());
+        if (!panel_ui->GetCurrentEntryId()) {
+          // Open/Close by tab change doesn't need animation.
+          // UI could refer this for preven animation.
+          operation_from_active_tab_change_ = true;
+        }
+
+        ActivatePanelItem(wanted_panel.value());
```

Comment:
If ActivatePanelItem() synchronously ends in the UI's change, could we use base::AutoReset instead of making ::IsPanelOperationFromActiveTabChange() reset the bit? I'm a little bit worried about having a getter reset the value. If we can't, I'd like to sugget that we change the method name more explicit. Like, 'GetIsPanelOperationFromActiveTabChangeAndReset()' even though it's mouthful